Client/src/Components/Dialog/index.jsx

```diff
@@ -0,0 +1,80 @@
+import LoadingButton from "@mui/lab/LoadingButton";
+import { Button, Modal, Stack, Typography } from "@mui/material";
+import PropTypes from "prop-types";
+const Dialog = ({ modelTitle, modelDescription,
+    open, onClose, title, confirmationBtnLbl, confirmationBtnOnClick, cancelBtnLbl, cancelBtnOnClick, theme, isLoading, description }) => {
+    return <Modal
+        aria-labelledby={modelTitle}
+        aria-describedby={modelDescription}
+        open={open}
+        onClose={onClose} >
+        <Stack
+            gap={theme.spacing(2)}
+            sx={{
+                position: "absolute",
+                top: "50%",
+                left: "50%",
+                transform: "translate(-50%, -50%)",
+                width: 400,
+                bgcolor: theme.palette.background.main,
+                border: 1,
+                borderColor: theme.palette.border.light,
+                borderRadius: theme.shape.borderRadius,
+                boxShadow: 24,
```

Comment:
I noticed that most of modal, and stack are equal throughout the project. 
I suggest the Dialog component becomes only <Modal props><Stack props> accepting the rest as children. 
That would decrease the amount of props we need to pass, avoid code repetitions (as is the case with the sx)

Reply:
One purpose I extract out the Dialog component is to not passing any styles when calling, it 
should all be enclosed within the component itself. If at any case, the Modal has special styles 
or element, it simply does not need to use Dialog component , use Modal directly and do the variations